### PR TITLE
appflowy: add desktop icon

### PR DESCRIPTION
**Pull request by blackheaven**
## Description of changes

I have defined the Icon based on official appflowy icon

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I have tested it with Plasma/KDE

### CODE REVIEW

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 275240` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>appflowy</li>
  </ul>
</details>

**Review by donovanglover — APPROVED** [review]
LGTM

File: pkgs/applications/office/appflowy/default.nix [hunk 1]
```diff
@@ -47,6 +47,10 @@ stdenv.mkDerivation rec {
     # Copy archive contents to the outpout directory
     cp -r ./* $out/opt/
 
+    # Copy icon
+    mkdir -p $out/share/icons
+    cp data/flutter_assets/assets/images/flowy_logo.svg $out/share/icons/appflowy.svg
+
```
Comment (peterhoeg) [inline pkgs/applications/office/appflowy/default.nix:53]: you can replace these 2 lines with:
```
install -Dm444 data/flutter_assets/assets/images/flowy_logo.svg $out/share/icons/appflowy.svg
```

Comment (blackheaven) [reply pkgs/applications/office/appflowy/default.nix:53]: Thanks

**Review by DarkOnion0 — APPROVED** [review]
App runs and builds well, LGTM!

---

Result of `nixpkgs-review pr 275240` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>appflowy</li>
  </ul>
</details>

File: pkgs/applications/office/appflowy/default.nix [hunk 2]
```diff
@@ -47,6 +47,9 @@ stdenv.mkDerivation rec {
     # Copy archive contents to the outpout directory
     cp -r ./* $out/opt/
 
+    # Copy icon
+    install -Dm444 data/flutter_assets/assets/images/flowy_logo.svg $out/share/icons/appflowy.svg
```
Comment (2xsaiko) [inline pkgs/applications/office/appflowy/default.nix:51]:
```suggestion
    install -Dm444 data/flutter_assets/assets/images/flowy_logo.svg $out/share/pixmaps/appflowy.svg
```

```suggestion
    install -Dm444 data/flutter_assets/assets/images/flowy_logo.svg $out/share/icons/hicolor/scalable/apps/appflowy.svg
```

AFAIK /usr/share/icons is for specific icon themes.

**Review by dblsaiko — CHANGES_REQUESTED** [review]

File: pkgs/applications/office/appflowy/default.nix (hunk 2 above)
Comment (blackheaven) [reply pkgs/applications/office/appflowy/default.nix]: I didn't know that, thanks.

For reference: https://specifications.freedesktop.org/icon-theme-spec/latest/ar01s03.html

File: pkgs/applications/office/appflowy/default.nix [hunk 3]
```diff
@@ -47,6 +47,10 @@ stdenv.mkDerivation rec {
     # Copy archive contents to the outpout directory
     cp -r ./* $out/opt/
 
+    # Copy icon
+    install -Dm444 data/flutter_assets/assets/images/flowy_logo.svg $out/share/pixmaps/appflowy.svg
+    install -Dm444 data/flutter_assets/assets/images/flowy_logo.svg $out/share/icons/hicolor/scalable/apps/appflowy.svg
```
Comment (2xsaiko) [inline pkgs/applications/office/appflowy/default.nix:52]: Sorry, I should have specified that was an alternative. You should only need one of these 😅

**Review by dblsaiko — CHANGES_REQUESTED** [review]

File: pkgs/applications/office/appflowy/default.nix (hunk 3 above)
Comment (blackheaven) [reply pkgs/applications/office/appflowy/default.nix]: Sorry, I'm tired :/

**Review by dblsaiko — APPROVED** [review]

File: pkgs/applications/office/appflowy/default.nix (hunk 3 above)
Comment (2xsaiko) [reply pkgs/applications/office/appflowy/default.nix]: Aren't we all 🫠

**Review by DarkOnion0 — APPROVED** [review]
Result of `nixpkgs-review pr 275240` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>appflowy</li>
  </ul>
</details>

**Pull request merged**